### PR TITLE
add unit test for ip_range

### PR DESCRIPTION
**Pull request by zcq98**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Tests

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

**Review by bobz965 — APPROVED** [review]

**Comment by bobz965** [comment]
@zhangzujian could you please help review? thanks!

File: pkg/ipam/ip_range_test.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+		})
+	}
+}
+
+func equalIPRanges(a, b []*IPRange) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if !a[i].Start().Equal(b[i].Start()) || !a[i].End().Equal(b[i].End()) {
```
Comment (zhangzujian) [inline pkg/ipam/ip_range_test.go:225]:
```suggestion
		if !reflect.DeepEqual(a[i], b[i]) {
```

File: pkg/ipam/ip_range_test.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+	end := IP(net.ParseIP("192.168.1.10"))
+	r := NewIPRange(start, end)
+	clone := r.Clone()
+
+	if !clone.Start().Equal(start) || !clone.End().Equal(end) {
+		t.Errorf("Clone() = %v-%v, want %v-%v", clone.Start(), clone.End(), start, end)
+	}
+
+	clone.SetStart(IP(net.ParseIP("10.0.0.1")))
+	if clone.Start().Equal(start) {
```
Comment (zhangzujian) [inline pkg/ipam/ip_range_test.go:53]:
```suggestion
	if r.Start().Equal(clone.Start()) {
```

File: pkg/ipam/ip_range_test.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, ok := r.Remove(tt.ip)
+			if ok != tt.wantOk {
+				t.Errorf("Remove(%v) ok = %v, want %v", tt.ip, ok, tt.wantOk)
+			}
+			if !equalIPRanges(got, tt.wantRanges) {
```
Comment (zhangzujian) [inline pkg/ipam/ip_range_test.go:213]: Can we use `reflect.DeepEqual` here?

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**